src/units/combine.rs

```diff
@@ -195,6 +195,58 @@ impl metrics::Source for AnyMetrics {
 }
 
 
+//------------ Merge ---------------------------------------------------------
+
+/// A unit merging the data sets of all upstream units.
+#[derive(Debug, Deserialize)]
+pub struct Merge {
+    /// The set of units whose data set should be merged.
+    sources: Vec<Link>,
+}
+
+impl Merge {
+    pub async fn run(
+        mut self, mut component: Component, mut gate: Gate
+    ) -> Result<(), Terminated> {
+        if self.sources.is_empty() {
+            gate.update(UnitUpdate::Gone).await;
+            return Err(Terminated)
+        }
+        let metrics = gate.metrics();
+        component.register_metrics(metrics.clone());
+
+        loop {
+            {
+                let res = select(
+                    select_all(
```

Comment:
This will wait for _any_ out of _all_ the sources, but it will not wait for all of them. Intuitively, I'd say you actually want the latter? (So, something more like a `join_all` perhaps?)

Reply:
Shouldn’t it update the merged set every time any of the units updates?